addons/purchase/views/purchase_views.xml

```diff
@@ -232,11 +242,11 @@
                                         name="product_id"
                                         widget="product_label_section_and_note_field"
                                         readonly="state in ('purchase', 'to approve', 'done', 'cancel')"
-                                        required="not display_type"
+                                        required="not display_type and not is_downpayment"
```

Comment:
this doesn't seem to be working + you should probably add a `is_downpayment` to the readonly condition

Reply:
I hesitated with the required since you could have a downpayment product, but in that case it should be used on the bill - so there shouldn't be any need to change it.

Reply:
doesn't seem to be working? seems ok to me... `product_id` is not required on downpayment lines